File: Java/Repetitivas/ej8.java

/**
 * Hacer un programa que muestre un cronometro, indicando las horas, minutos y segundos.
 * @author Alberto Miguel Martinez Jimenez
 *
 */
/**
 * Algoritmo
 * segundos = 0
 * minutos = 0
 * horas = 0
 * incremenar segundo cada 1000milisegundos 
 * incrementar minutos cada 60 segundos
 * poner segundos a 0
 * incrementar horas cada 60 minutos
 * poner a 0 minutos
 */
public class ej8 {

	public static void main(String[] args) {
		int segundos = 0;
		int minutos = 0;
		int horas = 0;
		
		while(true) {
			try{
				Thread.sleep(1000);
			}catch(InterruptedException e ) {
				}
			if(segundos!=59) {
				segundos++;
			}else{
				if(minutos!=59) {
					minutos++;
					segundos = 0;
				}else{
					horas++;
					minutos = 0;
				}
			}
			
			System.out.println(horas+":"+minutos+":"+segundos);
		}
	}
}

File: Python3/Repetitivas/ej8.py
# -*- coding: utf-8 -*-
import time
import os
#Hacer un programa que muestre un cronometro, indicando las horas, minutos y segundos.

#bucle muestra cada segundo la hora, minuto y segundo
#y limpia la pantalla

print("\/\/\/CRONOMETRO\/\/\/")
contador = 0
while True:
    #print(contador)
    #tiempo = time.gmtime(time.time())
    #print(tiempo.tm_hour,":",tiempo.tm_min,":",tiempo.tm_sec)
    print(horas,":",minutos,".",segundos)
    time.sleep(1)
    if  segundos < 59:
	segundos += 1
    else:
	segundo = 0
	minutos += 1
	
    if  minutos < 59:
	minutos += 1
    else:
	minutos = 0
	horas += 1
    contador = contador + 1
    if(os.name=="posix"): #Si el SO es linux o mac
        os.system("clear")
    elif(os.name=="nt"): #Si el SO es windows
        os.system("cls")
    print("\/\/\/CRONOMETRO\/\/\/")
   
	
	
